packages/graphiql/src/components/QueryEditor.tsx

```diff
@@ -47,227 +38,62 @@ type QueryEditorProps = {
-            props.onCopyQuery();
-          }
-        },
-
-        'Shift-Ctrl-P': () => {
-          if (props.onPrettifyQuery) {
-            props.onPrettifyQuery();
-          }
-        },
-
-        /* Shift-Ctrl-P is hard coded in Firefox for private browsing so adding an alternative to Pretiffy */
-        'Shift-Ctrl-F': () => {
-          if (props.onPrettifyQuery) {
-            props.onPrettifyQuery();
-          }
-        },
-
-        'Shift-Ctrl-M': () => {
-          if (props.onMergeQuery) {
-            props.onMergeQuery();
-          }
-        },
-        ...commonKeys,
-        'Cmd-S': () => {},
-        'Ctrl-S': () => {},
+    const editor = (editorRef.current = monaco.editor.create(
+      divRef.current as HTMLDivElement,
+      {
+        value: session?.operation?.text ?? '',
+        language: 'graphqlDev',
+        automaticLayout: true,
+        minimap: { enabled: false },
       },
-    });
-
-    const editHandler = () => _onEdit(session.changeOperation);
-    if (editorRef.current) {
-      editorRef.current.on('change', editHandler);
-      editorRef.current.on('keyup', _onKeyUp);
-      // @ts-ignore @TODO additional args for hasCompletion event
-      editorRef.current.on('hasCompletion', _onHasCompletion);
-      editorRef.current.on('beforeChange', _onBeforeChange);
+    ));
+    if (!editor) {
+      return;
     }
-
-    return function cleanup() {
-      if (editorRef.current) {
-        editorRef.current.off('change', editHandler);
-        editorRef.current.off('keyup', _onKeyUp);
-        // @ts-ignore
-        editorRef.current.off('hasCompletion', _onHasCompletion);
-        editorRef.current.off('beforeChange', _onBeforeChange);
+    loadEditor('operation', editor);
+    editor.onDidChangeModelContent(() => {
+      if (!ignoreChangeEvent) {
+        cachedValueRef.current = editor.getValue();
+        session.changeOperation(cachedValueRef.current);
       }
-    };
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   React.useEffect(() => {
+    setIgnoreChangeEvent(true);
     const editor = editorRef.current;
-    const CodeMirror = require('codemirror');
-    // Ensure the changes caused by this update are not interpretted as
-    // user-input changes which could otherwise result in an infinite
-    // event loop.
-    ignoreChangeEventRef.current = true;
-    if (editor && editor.options.lint.schema !== schema) {
-      // on schmea change, update the underlying schemas
-      editor.options.lint.schema = schema;
-      editor.options.hintOptions.schema = schema;
-      editor.options.info.schema = schema;
-      editor.options.jump.schema = schema;
-      CodeMirror.signal(editor, 'change', editor);
-    }
-
-    if (session?.operation?.text !== cachedValueRef.current && editor) {
-      cachedValueRef.current = session?.operation?.text ?? '';
-      editor.setValue(session?.operation?.text ?? '');
+    const op = session?.operation?.text;
```

Comment:
According to TypeScript `session.operation.text` is safe here, which would fire this affect less often
```suggestion
    const op = session.operation.text;
```